compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -838,6 +856,119 @@ runScriptsRule opts =
       let (diags, results) = unzip scenarioResults
       pure (catMaybes diags, Just results)
 
+runScriptsPkg ::
+       NormalizedFilePath
+    -> LF.ExternalPackage
+    -> [LF.ExternalPackage]
+    -> Action ([FileDiagnostic], Maybe [(VirtualResource, Either SS.Error SS.ScenarioResult)])
+runScriptsPkg projRoot extPkg pkgs = do
```

Comment:
`runScriptsPkg` and `runScenariosPkg` look almost identical and they should also be able to share the same context. Can we merge them into one function that runs both given that we never run them separately?

Reply:
merged the two functions.